libafl_qemu/src/qemu/qemu_opt.rs

```diff
@@ -0,0 +1,336 @@
+use core::fmt;
+use std::{fmt::Formatter, path::PathBuf};
+
+use strum_macros::Display;
+
+#[allow(non_camel_case_types)]
+#[derive(Default, Debug, Display, Clone, Copy)]
+pub enum QemuOptAccelerator {
+    kvm,
+    #[default]
+    tcg,
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Default, Debug, Display, Clone, Copy)]
+pub enum QemuOptDriveInterface {
+    floppy,
+    #[default]
+    ide,
+    mtd,
+    none,
+    pflash,
+    scsi,
+    sd,
+    virtio,
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, Display, Clone, Copy)]
+pub enum QemuOptDiskImageFileFormat {
+    qcow2,
+    raw,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct QemuOptDrive {
+    file: Option<PathBuf>,
+    format: Option<QemuOptDiskImageFileFormat>,
+    interface: Option<QemuOptDriveInterface>,
+}
+
+impl fmt::Display for QemuOptDrive {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, " -drive")?;
+
+        let mut is_first_option = true;
+        let mut separator = || {
+            if is_first_option {
+                is_first_option = false;
+                " "
+            } else {
+                ","
+            }
+        };
+
+        if let Some(file) = &self.file {
+            write!(f, "{}file={}", separator(), file.to_str().unwrap())?;
+        }
+        if let Some(format) = &self.format {
+            write!(f, "{}format={}", separator(), format)?;
+        }
+        if let Some(interface) = &self.interface {
+            write!(f, "{}if={}", separator(), interface)?;
+        }
+
+        Ok(())
+    }
+}
+
+#[allow(non_camel_case_types)]
+#[derive(Default, Debug, Display, Clone, Copy)]
+pub enum QemuOptSerial {
+    none,
+    null,
+    #[default]
+    stdio,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct QemuOpt {
```

Comment:
did you have a look at [TypedBuilder](https://docs.rs/typed-builder/latest/typed_builder/index.html) or maybe [clap](https://docs.rs/clap/latest/clap/)?
i think a lot of boilerplate code can be removed by using the right crate.

Reply:
Okk, the new version uses TypeBuilder. But since the `Qemu` struct cannot be used to store the config, the result is a little janky